mach-eight/reup.romulo
# Release Sept 06 2024 (#193)

* Update Building data model

* Comments addressed

* Fix indentation

* Change versiont from 1.1.2 to 1.1.3

* Address comments

* Address comments

* move insertion object

* Rename PointSensor

* Create Object Sensor

* Add test of dependency injections and highlight one object

* Hightlight same object only once

* unhighlight object

* Test object selector is for selectable objects

* Add switch to sensedobjectHighlighter

* make object sensor not a monobehaviour

* create test to enable highlight selectable objects only in edit mode

* do not highlight selected objects

* Create ReupPrefabInstantiator for test purposes

* Apply ReupPrefabInstantiator in Maintain height tests

* rename ReupSceneInstantiator

* create metada in registered identifier

* Save color metadata when changing object color

* object mapper maps metadata

* Move helper functions

* Fix identation

* create GetSceneState method

* create scene state message implementation

* Assign Material id MetaData

* update build version

* request material id in message

* receive and reply with scene name

* Add support for required fields in DataValidation

* add required fields to External schema

* Add required fields to internal schema

* Validate internal schemas

* Support schema references

* Fix internal schemas

* make generic meta data utils

* Deselect objects when sending scene state

* change border colors

* clean code

* change back api url

* Fix

* Update Version

* change tag data model

* rename and create new actions

* create Tag str method

* update ignore layer

* fix test

* accept null schema and Multi schema

* simplify CreateArraySchema call

* Add more tests

* paint objects to load

* Update version to 1.1.5

* Tool to assign tags to several objects at the same time

* make first test to apply materials to load objects

* Test several materials in load scene

* await for material controller task

* Send success message

* rename actions namespace

* Fix receiving nulls in request scene load message

* save color with #

* rename actions

* Comments Addressed

* Comments Addressed

* Comments Addressed

* Comments Addressed

* version 1.1.6

* Deselect all items

* Update Version

* save original material in object info

* create OriginalSceneController

* Call Restore original scene from edition mediator

* populating object info

* set material change info when changing color

* Create ObjectInfoController

* update build version to 2024 07 24

* add object info component at build time

* remove object info controller from building setup dependency injector

* Revert "remove object info controller from building setup dependency injector"

This reverts commit 35733183d31a8ef35365d53b0444b50f3ec37340.

* Rename EditionMediator to EditMediator

* Rename EditionMediator to EditMediator

* SubEdit Mode Unity implementation

* Fix tests

* return all objects with that passed inverted filters

* Comments Addressed

* Refactor

* Comments addressed

* update to version 1.1.8

* Comments Addressed

* Update version to 1.1.9 and buildVersion to "2024-08-01"

* also remove object info from tool

* update trilib used api

* Fix tests

* update test unity version

* add text mesh pro as dependency

* add visual scripting dependency

* Create halo applier

* downgrade to 2022.3.15

* Remove Asset

* remove renderers folder

* remove sample scene

* Remove unnused Asset

* create ISelectableObjectsHighlighter

* add test to find selectable objects highlighter

* implement HioghlighAnimator in SelectedObjectsManager

* implement Highligh animator logic

* Add Dummy emission object

* add Highlight animator to prefab

* Revert "Remove URP manual assets"

* Add new files after TriLib update

* stop old highlight before new one

* rename methods

* Update version to 1.1.10 and buildVersion to "2024-08-13"

* Change requestSceneStatePayloadSchema and update message send by editMediator

* Update test with new schema

* Add error log in receive web message method

* move method

* Update validator schema

* fix tests

* scale material when applying it to surface

* clean code

* test to reduce matrix

* create test for UvUtils

* implement MaterialScaler

* clean UvUtils

* rename method

* test MaterialScaler

* rename test

* check if renderer exist first in UvUtils

* defer conversion to meters

* fix tests

* import external dll to do reduced row echelon form algorithm

* fix selecting collinear points

* Update Runtime/Helpers/UvUtils.cs

* updating validator schema

* fix typos

* fix where, schema and test

* fix test

* Change data validator schemas to camel case and fix tests

* update materialId metadata path and tests

* unify material schemas and fix edit mediator test

* Fix change material controller

* update unity version

* first test of object visibility utils

* second test

* Apply Visibility

* User utils in tags scanner tool

* clean

* update renderer

* search tags by name and id

* select first non collinear triangle

* make tests

* Use select instead of for

* update to version 2.0.1

* Json Schema third party solution, code refactor and fix tests

* Comments addressed

* Implement states history to undo in order

* rename undo button

* Address comments

* Comments addressed

* fix indentation

* dispose downloadHandler after is used

* compress downloaded texture to ASTC_12x12

* set ASTC compression format when building

* Fix tests

* fix if order

* Comments addressed

* fix indentation

* Fix asset

* move ObjectVisibilityUtils to editor folder

* organize Tests to property have edit mode tests

* fix assembly definition

* separate mocks

* Change milimeters for millimeters

* update to version 2.0.2

---------
